components/Navigation.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div class="flex ml-auto gap-10 m-2 mr-20 space-x-1 text-l">
+    <div
+      v-for="(page, i) in pages"
+      :key="i"
+      class="space-y-0.5"
+      @mouseover="page.hover = true"
+      @mouseleave="page.hover = false"
+    >
+      <!-- main button -->
+      <NuxtLink :to="page.parent.url">
+        <button
+          class="w-32 bg-blue-300 hover:bg-blue-500 text-white py-2 px-4 rounded"
+        >
+          {{ page.parent.title }}
+        </button>
+      </NuxtLink>
+      <!-- dropdown pages -->
+      <div
+        v-if="page.hover"
+        class="absolute flex flex-col space-y-0.5"
+      >
+        <div
+          v-for="subPage in page.children"
+          :key="subPage"
+        >
+          <NuxtLink :to="subPage.url">
+            <button class="w-32 bg-blue-300 hover:bg-blue-500 text-white py-2 px-4 rounded">
+              {{ subPage.title }}
+            </button>
+          </NuxtLink>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  data () {
+    return {
+      hover: false,
+      pages: [
+        {
+          parent: { title: 'CPM', url: '/cpm' },
+          children: [{ title: 'Explore', url: '/cpm/explore' },
+            { title: 'Analyse', url: '/cpm/analyse' },
+            { title: 'Download', url: '/cpm/download' },
+            { title: 'More info', url: '/cpm/references' }],
+          hover: false
+        },
+        {
+          parent: { title: 'Decadal', url: '/decadal' },
+          children: [{ title: 'Forecasts', url: '/decadal/forecasts' },
+            { title: 'Hindcasts', url: '/decadal/hindcasts' },
+            { title: 'Experiments', url: '/decadal/experiments' }],
+          hover: false
```

Comment:
Separate functionality from data: `hover` is a bit out of place here, and you need to type it every time and always fill in false. What about:

```
export default {
  data () {
    pages = [ ... ]
    pages.forEach(page => page.hover = false)
    return pages
  }
}
```

Still not ideal I'd say, but at least now I can easily modify the data without having to worry about the way it is used internally.

Reply:
Nice suggestion. But it seems this syntax is not allowed. I added `this.pages = this.pages.map(page => ({ ...page, hover: false }))` in `mounted () {}` and it works well now.